openedx/core/djangoapps/content/block_structure/config/__init__.py

```diff
@@ -10,12 +10,51 @@
 from .models import BlockStructureConfiguration
 
 # Switches
+# .. toggle_name: block_structure.invalidate_cache_on_publish
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: When enabled, the block structure cache is invalidated when changes to
+#   courses are published. Enabling this toggle is useful for non-expiring block structure cache.
+#   It is also useful to enable this when `block_structure.storage_backing_for_cache` is enabled
+#   because it guarantees the storage, containing the storage-backed structures, is updated with
+#   the latest changes.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14571
 INVALIDATE_CACHE_ON_PUBLISH = WaffleSwitch(
     "block_structure.invalidate_cache_on_publish", __name__
 )
+
+# .. toggle_name: block_structure.storage_backing_for_cache
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: When enabled, block structures are stored in a more permanent storage,
+#   like a database, which provides an additional backup for cache misses, instead having them
+#   regenerated. The regenration of block structures is a time consuming process. Therefore,
+#   enabling this switch is recommended for Production.
+#   For an example with additional context, see the following forum post:
+#   https://discuss.openedx.org/t/help-please-very-slow-load-time-10-seconds-for-courseware-on-sections-with-several-subsections-and-xblocks/2998/16  pylint: disable=line-too-long
```

Comment:
Ugh. Now it is `36: I0021: (useless-suppression): Useless suppression of 'line-too-long'`. Hopefully we don’t have too linters playing tug-of-war. Make note of what script is failing, in case a different fails if you add useless-suppression. I fear there may not be a combo that makes it happy, but you’ll see. Good luck. If it doesn’t work out, we can do the multi-line url and I’ll let others know about the problem.

Thank you!

Reply:
Apparently, adding `useless-suppression` to that pragma did not cause additional issues \o/ 🥳 

I'm going to squash, again 😃

Reply:
That’s good. But why did we need it at all? Frustrating, but at least it works.
FYI: @jmbowman: Odd issue with pylint violations that failed on line too long, and then complained when we disabled it.

Reply:
@robrap I think it considered the whole line a pragma, possibly. I did a quick search about any similar issues, I didn't find anything, but there's a chance I might if I spend more time on it.

But from a glance, it seems like when `pylint: ` was added, it might have considered the whole comment as a pragma. I believe they are not considered when the line length is checked, which is what happened here.